app/controllers/settings/applications_controller.rb

```diff
@@ -61,11 +60,9 @@ def set_application
 
   def application_params
     params
-      .expect(doorkeeper_application: [:name, :redirect_uri, :scopes, :website])
-  end
-
-  def prepare_scopes
-    scopes = application_params.fetch(:doorkeeper_application, {}).fetch(:scopes, nil)
-    params[:doorkeeper_application][:scopes] = scopes.join(' ') if scopes.is_a? Array
+      .expect(doorkeeper_application: [:name, :redirect_uri, :website, scopes: []])
+      .tap do |params|
+      params[:scopes] = params[:scopes]&.join(' ')
```

Comment:
Apologies for missing this one ... I was on the look out for nested attributes and missed this array.

I wonder if we could move the "accept an array but save a string" portion to a `normalizes` on the model? Something like `Array(value).join(' ')` (maybe a `compact` in there too) would handle both the inbound string and inbound array scenarios (I think?), and might make this controller portion less odd.